src/providers/azurestack/mod.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a stub provider. Right now AzureStack does absolutely nothing
+// interesting.
+
+//! azurestack/azurestack metadata fetcher
+use std::collections::HashMap;
+
+use crate::errors::*;
+use crate::network;
+use crate::providers::MetadataProvider;
+
+use openssh_keys::PublicKey;
+use slog_scope::warn;
+
+#[derive(Clone, Copy, Debug)]
+pub struct AzureStack;
+
+impl AzureStack {
+    pub fn new() -> Self {
+        Self
+    }
+
+    /// Get the hostname from local system settings.
+    fn system_hostname() -> Result<String> {
+        let hostname = hostname::get()
+            .chain_err(|| "unable to get hostname")?
+            .to_string_lossy()
+            .into_owned();
+        Ok(hostname)
+    }
+}
+
+impl MetadataProvider for AzureStack {
+    fn attributes(&self) -> Result<HashMap<String, String>> {
+        let hostname = Self::system_hostname()?;
```

Comment:
I don't think it is useful to pipe back the local hostname here, we can plainly drop this.